src/libbson/src/bson/bson-types.h

```diff
@@ -519,6 +519,40 @@ typedef struct _bson_error_t {
 } bson_error_t BSON_ALIGNED_END (8);
 
 
+/**
+ * BSON_MAX_LEN_UNLIMITED
+ *
+ * Denotes unlimited length limit when converting BSON to JSON.
+ */
+#define BSON_MAX_LEN_UNLIMITED -1
```

Comment:
I'm inclined to say that `-1` is more in-line with other driver functions that deal with string length, like the `bson_append_*` functions that take `-1` to mean "no specified key length."  Here, it would be kind of parallel, "no specified length limit," or unlimited.  Though I'm not sure why you would use it, I would also expect setting `0` for the max length to mean empty logging strings, and I think it would be a little counterintuitive if `0` meant "unlimited."